src/LAMMPS.jl

```diff
@@ -168,20 +169,15 @@ mutable struct LMP
     external_fixes::Dict{String, Any}
 
     function LMP(args::Vector{String}=String[], comm::MPI.Comm=MPI.COMM_WORLD)
-        args = copy(args)
-        pushfirst!(args, "lammps")
-
-        GC.@preserve args begin
-            if API.lammps_config_has_mpi_support() == 0
-                handle = API.lammps_open_no_mpi(length(args), args, C_NULL)
-            else
-                if !MPI.Initialized()
-                    error("MPI has not been initialized. Make sure to first call `MPI.Init()`")
-                end
-                handle = API.lammps_open(length(args), args, comm, C_NULL)
-            end
+        MPI.Initialized() || MPI.Init()
+        if API.lammps_config_has_mpi_support() == 0 && MPI.Comm_size(comm) != 1
+            msg = "The lammps binary doesn't support MPI but the communicator has size > 1.\n" *
```

Comment:
So this is actually a situation that might be intentional, using MPI to run many separate lammps simulations for parameter exploration comes to mind.